### PR TITLE
Invoke task to build otel-agent binary

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Adds an invoke task to build the otel-agent

### Motivation

Can build with: `invoke otel-agent.build`

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

File: tasks/otel_agent.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    """
+
+    if os.path.exists(BIN_PATH):
+        os.remove(BIN_PATH)
+
+    env = {"GO111MODULE": "on"}
+    build_tags = ['otlp']
+
+    cmd = 'go build -tags="{go_build_tags}" '
+    cmd += '-o {agent_bin} {REPO_PATH}/cmd/otel-agent'
```
Comment (chouetz) [inline tasks/otel_agent.py:24]:
```suggestion
    cmd = f"go build -tags=\"{' '.join(build_tags)}\" -o {BIN_PATH} {REPO_PATH}/cmd/otel-agent"
```
I'm not sure of the reason why you split this. And maybe for readability I'd suggest using f-strings

Comment (KSerrania) [reply tasks/otel_agent.py:24]: I'd say it's a copy-paste from the [agent build task](https://github.com/DataDog/datadog-agent/blob/610e0ce8837a8512dda19f54c21913bf666e718f/tasks/agent.py#L198-L218), which is using this form (and hasn't been updated in years).

Comment (dustmop) [reply tasks/otel_agent.py]: > I'd say it's a copy-paste from the [agent build task](https://github.com/DataDog/datadog-agent/blob/610e0ce8837a8512dda19f54c21913bf666e718f/tasks/agent.py#L198-L218), which is using this form (and hasn't been updated in years).

That's correct. Pushed a commit to update it here as recommended.

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**